### DISCUSSION TITLE
how to install the update without losing all the personal settings

### DISCUSSION THREAD

**Discussion by faon-92** [SubtitleEdit/subtitleedit#7088] [General]
Hi everyone,
I made many personal settings and everytime I install an update, I export the shortcuts then without desinstall the existing one, I install the new version just over it, then import the shortcuts. Is it the good procedure?

I'm wondering that all the personal setting concerning other sections of the settings could be preserved in this way?

I have the odd impression that sometimes they are preserved, sometimes not...

thank you for your insights,

**Comment by GrampaWildWilly** [comment]
When I get a new version of SE -- note, I get it, I don't install it -- I have gotten in the habit of test-unzipping the package.  Nik told me a long time ago that the SE settings are in a file named Settings.xml.  One time when I had the annoying occurrence of losing all my settings when I got a new SE, he apologized to me for accidentally including a Settings.xml file in the SE distribution package.  He said he makes an effort to make sure that the SE packages do NOT include a Settings.xml file.  If there is no such file in the SE directory when you first execute SE, it constructs the file.  If such a file already exists, SE just uses it.  I'm sure there are times when SE adds certain things to Settings.xml, & other times when it removes them.  But that's all magic that Nik does under the covers & we shouldn't be concerned with that.  What you describe, saving Settings.xml, putting a new SE into place, then restoring Settings.xml, is certainly a prudent thing to do, but not entirely necessary . . . except in the rare cases when Nik forgets.  I have to say that since that one time when I pointed it out to him, every SE I've gotten has NOT included a Settings.xml.  It's been a couple of years now.  Still, I've made it a habit to check.  If the zip file should some day accidentally include a Settings.xml, I will just remove it from the zip archive before I unzip it into its home directory.  And I'll drop Nik a note about it.  I'm sure he'll repair the file here pretty quickly.

I said I have not installed SE.  When I go into Windows Control Panel & go to the function for uninstalling installed software, SE is not listed on my system.  I just unzip the file you can get from here (GitHub) into a directory somewhere on my disk space.  That directory is not on my boot partition.  I long ago created a shortcut to SubtitleEdit.exe in that directory.  That shortcut has been on my Start Menu for so long, I can't even remember when I did that.  It's been at least 3 years.  The most I do is change the name of the shortcut to include whatever the number is of the latest beta, or the release number when Nik gives us a new release.  But that's just me being AR/OCD.  It's not entirely necessary.  I hardly ever run an actual release for very long.  Any time I'm going to do anything with captions, I first come here & grab the latest beta.  I figure if we're going to talk about SE here, we should at least be sure we're not hitting something Nik has already fixed.  He's established a track record of giving us betas that work, or at least mostly work.  So I no longer think twice about getting the latest beta.

**Comment by faon-92** [comment]
Thank you for your answer.
When I unzip the lastest version, there is only one file SubtitleEdit-3.6.13-Setup.exe inside. Maybe things changed over time?
Besides, you are talking about Settings.xm, not SE_Shortcuts.xml, is it correct?

**Comment by GrampaWildWilly** [comment]
Hmmm......  I do not have a file named SubtitleEdit-3.6.13-Setup.exe, just SubtitleEdit.exe.  But like I said, I don't actually install SE.  I just unzip it into a directory over top of whatever is already there.  That gets me whatever new code Nik & the other contributors here have written.  Plus, I pretty much am always running a beta.  There is no installer for the beta.

As for shortcuts, I assume you are talking about the things you define on the Shortcuts page of the Settings dialog.  Those things are all in the Settings.xml file.  I don't have a file named SE_Shortcuts.xml.  How did you come to have such a file?  Did Nik tell you to create it?  Or does it just appear when you run the installer?  I can say that I've defined a modest number of shortcuts & I can see them when I edit the Settings.xml file, which is something I don't usually do.  So saving Settings.xml includes preserving your shortcuts.  Plus, my shortcuts work without this SE_Shortcuts.xml file you say you have.  I've been using SE on & off for maybe 3 or 4 years now & I've never seen that file.  Maybe it existed further back in history.  Somebody else will have to address that possibility.

**Comment by faon-92** [comment]
I choose to export the shortcuts, it's an option
<img width="669" alt="SubtitleEdit_qbCdosoJ4S" src="https://github.com/SubtitleEdit/subtitleedit/assets/127494540/b0f83673-da80-4933-a44a-3fdc20ec0de4">

**Comment by faon-92** [comment]
I understand what you mean now.
Don't install the new version, but replace the SubtitleEdit.exe file inside the directory by the new one

**Comment by GrampaWildWilly** [comment]
Well, we seem to be teaching each other new things.  I had no idea you could export your shortcuts.  Interesting.  But the shortcuts are also in Settings.xml.  At least, they are in mine.  You should check yours.  I should think it would be simpler, not to mention more complete, to save your Settings.xml file & not bother exporting your shortcuts.

As for replacing files, there's more to the SE package than just SubtitleEdit.exe.  There's a total of 5 directories & 19 files in the latest beta package.  The 5 directories combine for a total of 381 additional files.  I would not simply replace the one file & say I'm done.  But simply replacing all those files is what I do.  I think at the very beginning I ran the SE installer.  But the first time I used a beta, I saw there was no installer, so I uninstalled the SE I already had & I've been simply unzipping SE without installing it since.  I remember exchanging several E-mails with Nik in which he was trying to talk me into using a particular beta because it solved some problem I was having, I forget what now.  I think I had done something with the captions for 1 or 2 videos at that point & was just beginning to get an idea of what I was dealing with.  So I was extremely hesitant about running a beta of something I'd been using for barely a week.  I look back now & realize I was just being a fraidy cat.  Makes me laugh.  Not that I'm any kind of seasoned veteran with this.  I just fix the captions on operas I download.  I'm not doing any fancy stuff like translating or automatically extracting captions from the audio, like all this other activity in here with Whisper & OpenAI & all that stuff that I have no clue what those guys are talking about.  I'm just a dabbler, not an expert.  I know a few things & I don't know a huge number of things.

**Comment by faon-92** [comment]
Thank you for sharing your experiences:)

Could someone confirm which one is the best procedure to keep all the personal setting: 
- install the new version over the old one by executing the subtitleedit.exe, 
or
- unzip the zip archive than simplement replace subtitleedit.exe by the latest one in the directory?

**Comment by GrampaWildWilly** [comment]
You will keep all your personal settings either way.  As long as Nik keeps his streak going & avoids including a Settings.xml file in the SE distribution package, either way of running will preserve your personal settings.  But you will get a damaged SE if you unzip the zip file but copy just the one file SubtitleEdit.exe from the zip archive to your SE directory.  That is a recipe for disaster.  You must unzip EVERY FILE in the zip archive into your SE directory.  You must replace SubtitleEdit.exe AND EVERY OTHER FILE AS WELL.  It's not just a single file that you replace.  You have to replace ALL of them.  But since the zip file is NOT supposed to include Settings.xml, that file with your personal settings will remain unchanged by the unzip.  If you're worried about it, definitely make a safe copy of Settings.xml.  That's probably a good idea no matter how you install.  But you must understand that you are supposed to unzip the ENTIRE zip archive into your SE directory, not just SubtitleEdit.exe.  One file is not enough.  You have to do ALL of them.

**Comment by faon-92** [comment]
this question has been already discussed and solved:
https://github.com/SubtitleEdit/subtitleedit/issues/2776
